libcloud/container/drivers/lxd.py

```diff
@@ -0,0 +1,1269 @@
+            req = '/%s/operations/%s/wait?timeout=%s' % (self.version,
+                                                         id,
+                                                         timeout)
+            response = self.connection.request(req)
+        except BaseHTTPError as e:
+
+            message_list = e.message.split(",")
+            message = message_list[0].split(":")[-1]
+
+            # if not found assume the operation completed
+            if message != '"not found"':
+                # something is wrong
+                raise LXDAPIException(message=e.message)
+
+        response_dict = response.parse_body()
+        assert_response(response_dict=response_dict, status_code=200)
+
+        # return a dummy container
+        container = Container(driver=self, name=container.name,
+                              id=container.name,
+                              state=ContainerState.TERMINATED,
+                              image=None, ip_addresses=[],
+                              extra=None)
+        return container
+
+    def list_containers(self, image=None, cluster=None, detailed=True):
+        """
+        List the deployed container images
+
+        :param image: Filter to containers with a certain image
+        :type  image: :class:`.ContainerImage`
+
+        :param cluster: Filter to containers in a cluster
+        :type  cluster: :class:`.ContainerCluster`
+
+        :param detailed
+
+        :rtype: ``list`` of :class:`.Container`
+        """
+
+        result = self.connection.request('/%s/containers' % self.version)
+        result = result.parse_body()
+
+        # how to treat the errors????
+        assert_response(response_dict=result, status_code=200)
+
+        meta = result['metadata']
+        containers = []
+        for item in meta:
+            container_id = item.split('/')[-1]
+            if not detailed:
+
+                container = Container(driver=self, name=container_id,
+                                      id=container_id,
+                                      image=image, ip_addresses=[],
+                                      extra={})
+            else:
+                container = self.get_container(id=container_id)
+            containers.append(container)
+
+        return containers
+
+    def ex_get_image(self, fingerprint):
+        """
+        Returns a container image from the given image fingerprint
+
+        :param fingerprint: image fingerprint
+        :type  fingerprint: ``str``
+
+        :rtype: :class:`.ContainerImage`
+        """
+        req = '/%s/images/%s' % (self.version, fingerprint)
+        response = self.connection.request(req)
+
+        #  parse the LXDResponse into a dictionary
+        response_dict = response.parse_body()
+        assert_response(response_dict=response_dict, status_code=200)
+
+        return self._to_image(metadata=response_dict['metadata'])
+
+    def install_image(self, path, **description_dict):
```

Comment:
Ideally, all those non standard arguments would be prefixed with ``ex_`` or you would take  a single ``ex_description_dict`` argument which is a dictionary.